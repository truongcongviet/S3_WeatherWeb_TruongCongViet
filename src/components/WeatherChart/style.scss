.weatherChart {
  height: 40vh;
  position: relative;
  overflow-y: hidden;
  height: 40%;
  width: 100%;

  .noteText {
    font-size: 20px;
    padding: 5px;
    position: fixed;
    width: 100%;
    z-index: 1;
  
  }
}
