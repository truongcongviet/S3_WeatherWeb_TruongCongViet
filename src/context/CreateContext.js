import { createContext, useCallback, useContext, useReducer, Fragment } from 'react';
import { weatherReducer, initialState } from './reducers/weatherReducer'
const CreateContext = createContext({});

export const useAppContext = () => {
    const { data, setData } = useContext(CreateContext)
    const dispatch = useCallback(
        ({ type, data }) => {
            setData({ data, type })
        },
        [setData]
    )
    return [data, dispatch]
}

export const WeatherProvider = props => {
    const { children } = props;
    const [data, dispatch] = useReducer(weatherReducer, initialState);
    return (
        <CreateContext.Provider value={{ data, setData: dispatch }}>
            <Fragment>
                {children}
            </Fragment>

        </CreateContext.Provider>
    )
}

export default CreateContext;