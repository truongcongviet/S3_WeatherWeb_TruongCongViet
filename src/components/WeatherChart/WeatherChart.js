import { XAxis, Tooltip, AreaChart, Area, ResponsiveContainer, YAxis } from "recharts";
import { CustomizedMarker, CustomizeActiveIcon, } from "../../helper/customizedChart";
import './style.scss';
import { useAppContext } from "../../context/CreateContext";

const WeatherChart = () => {
  const [data, dispatch] = useAppContext();
  return (
    <div className="weatherChart">

      <div className="noteText" data-testid="showInfo">
        <span className="tide-text">Tide</span>
        <span className="text-secondary"> â€¢ </span>
        <span className="text-warning">Sunrise and Sunset</span>
        <div className="text-center text-muted">{data.weather && data.weather.date1}</div>
      </div>
      <ResponsiveContainer width={2500} height='100%'>
        {data.virtualData ? (
          <AreaChart
            data={data.virtualData}
            margin={{
              top: 0,
              left: 0,
              bottom: 0,
            }}
          >
            <Tooltip content={<CustomizedMarker />} wrapperStyle={{ zIndex: 2 }} />

            <Area
              type="monotone"
              dataKey="tide"
              fill="var(--stellblue)"
              activeDot={false}
            />

            <Area
              type="monotone"
              dataKey="tempFake"
              stroke="var(--orange)"
              activeDot={<CustomizeActiveIcon />}
              fill="none"
              yAxisId="temp"
  
            />

            <YAxis yAxisId="temp" width={0} ticks={[0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60]}/>

            <YAxis width={0}
              ticks={[0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5]}
            />
            <XAxis dataKey="hour" />
          </AreaChart>
        ) : (
          "No data chart"
        )}
      </ResponsiveContainer>
    </div>
  )
}

export default WeatherChart;