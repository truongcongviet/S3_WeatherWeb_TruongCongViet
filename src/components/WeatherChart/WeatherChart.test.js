import React from 'react';
import WeatherChart from './WeatherChart';
import {render, screen, cleanup} from '@testing-library/react';
import { mount, configure  } from 'enzyme';
import toJson from 'enzyme-to-json';
import { WeatherProvider } from '../../context/CreateContext'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const wrapper = mount(<WeatherProvider> <WeatherChart /> </WeatherProvider>);
describe('ChartWeather Component', () => {
    test("expect to render ChartWeather component", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const chartWeatherComponent = wrapper.find(".weatherChart");
        expect(chartWeatherComponent.length).toBe(1);;
    });
})