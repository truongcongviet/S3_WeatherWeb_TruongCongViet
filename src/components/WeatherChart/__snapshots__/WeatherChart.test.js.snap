// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartWeather Component expect to render ChartWeather component 1`] = `
<WeatherProvider>
   
  <WeatherChart>
    <div
      className="weatherChart"
    >
      <div
        className="noteText"
        data-testid="showInfo"
      >
        <span
          className="tide-text"
        >
          Tide
        </span>
        <span
          className="text-secondary"
        >
           â€¢ 
        </span>
        <span
          className="text-warning"
        >
          Sunrise and Sunset
        </span>
        <div
          className="text-center text-muted"
        />
      </div>
      <ResponsiveContainer
        debounce={0}
        height="100%"
        width={2500}
      >
        <ResizeDetector
          handleHeight={true}
          handleWidth={true}
          onResize={[Function]}
          targetRef={
            Object {
              "current": <div
                class="recharts-responsive-container"
                style="width: 2500px; height: 100%;"
              />,
            }
          }
        >
          <div
            className="recharts-responsive-container"
            style={
              Object {
                "height": "100%",
                "maxHeight": undefined,
                "minHeight": undefined,
                "minWidth": undefined,
                "width": 2500,
              }
            }
          >
            <AreaChart
              barCategoryGap="10%"
              barGap={4}
              data={
                Array [
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 2.8,
                  },
                  Object {
                    "tide": 2.5,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 1.5,
                  },
                  Object {
                    "tide": 1,
                  },
                  Object {
                    "tide": 0.7,
                  },
                  Object {
                    "tide": 1,
                  },
                  Object {
                    "tide": 1.5,
                  },
                  Object {
                    "tide": 3,
                  },
                  Object {
                    "tide": 5,
                  },
                  Object {
                    "tide": 4.1,
                  },
                  Object {
                    "tide": 2.2,
                  },
                  Object {
                    "tide": 2.8,
                  },
                  Object {
                    "tide": 2.5,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 1.5,
                  },
                  Object {
                    "tide": 0.8,
                  },
                  Object {
                    "tide": 0.1,
                  },
                  Object {
                    "tide": 1.2,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 2.5,
                  },
                  Object {
                    "tide": 3.1,
                  },
                  Object {
                    "tide": 3,
                  },
                  Object {
                    "tide": 4,
                  },
                  Object {
                    "tide": 3,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 1.5,
                  },
                  Object {
                    "tide": 1.2,
                  },
                  Object {
                    "tide": 0.7,
                  },
                  Object {
                    "tide": 1,
                  },
                  Object {
                    "tide": 1.5,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 2.5,
                  },
                  Object {
                    "tide": 2.8,
                  },
                  Object {
                    "tide": 2.9,
                  },
                  Object {
                    "tide": 2.8,
                  },
                  Object {
                    "tide": 2.5,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 1.5,
                  },
                  Object {
                    "tide": 1,
                  },
                  Object {
                    "tide": 0.7,
                  },
                  Object {
                    "tide": 1,
                  },
                  Object {
                    "tide": 1.5,
                  },
                  Object {
                    "tide": 2,
                  },
                  Object {
                    "tide": 2.5,
                  },
                  Object {
                    "tide": 2.8,
                  },
                ]
              }
              height={0}
              layout="horizontal"
              margin={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "top": 0,
                }
              }
              reverseStackOrder={false}
              stackOffset="none"
              syncMethod="index"
              width={2500}
            />
          </div>
        </ResizeDetector>
      </ResponsiveContainer>
    </div>
  </WeatherChart>
   
</WeatherProvider>
`;
